test/integration/apis/redeem.test.ts

```diff
@@ -56,81 +64,84 @@ describe.skip("redeem", () => {
+            btcCore.initializeClientConnection("regtest", "0.0.0.0", "rpcuser", "rpcpassword", "18443", "Alice");
+            const blocksToMine = 3;
+            keyring = new Keyring({ type: "sr25519" });
+            alice = keyring.addFromUri("//Alice");
+            dave = keyring.addFromUri("//Dave");
 
-            const listingsPerPage = 2;
-            const requestsIterator = redeemAPI.getPagedIterator(listingsPerPage);
-            let curr = await requestsIterator.next();
-            let requestCount = 0;
-            while (!curr.done) {
-                requestCount += curr.value.length;
-                assert.isTrue(curr.value.length <= listingsPerPage);
-                curr = await requestsIterator.next();
-            }
-            assert.equal(requestCount, sentRequests);
-        });
-    });
+            // request issue
+            issueAPI.setAccount(alice);
+            const amountAsBtcString = "0.1";
+            const amountAsSatoshiString = btcToSat(amountAsBtcString);
+            const amountAsSatoshi = api.createType("Balance", amountAsSatoshiString);
+            const requestResult = await issueAPI.request(amountAsSatoshi);
+            assert.isTrue(requestResult.hash.length > 0);
 
-    describe("execute", () => {
-        it("should fail if no account is set", () => {
-            const redeemId: H256 = <H256>{};
-            const txId: H256Le = <H256Le>{};
-            const txBlockHeight: u32 = <u32>{};
-            const merkleProof: Bytes = <Bytes>{};
-            const rawTx: Bytes = <Bytes>{};
-            assert.isRejected(redeemAPI.execute(redeemId, txId, txBlockHeight, merkleProof, rawTx));
-        });
+            // send btc tx
+            const data = stripHexPrefix(requestResult.hash.toString());
+            const vaultBtcAddress = encodeBtcAddress(requestResult.vault.wallet.address, bitcoin.networks.regtest);
+            if (vaultBtcAddress === undefined) {
+                throw new Error("Undefined vault address returned from RequestIssue");
+            }
+            const txData = await btcCore.broadcastOpReturnTx(vaultBtcAddress, amountAsBtcString, data);
+            await btcCore.mineBlocks(blocksToMine);
+            await delay(10000);
 
-        it("should request if account is set", async () => {
+            // redeem
             redeemAPI.setAccount(alice);
-            const amount = api.createType("PolkaBTC", 10);
-            requestResult = await redeemAPI.request(amount, randomDecodedAccountId);
+            const redeemAmountAsBtcString = "0.09";
+            const redeemAmountAsSatoshiString = btcToSat(redeemAmountAsBtcString);
+            const redeemAmountAsSatoshi = api.createType("Balance", redeemAmountAsSatoshiString);
+            const btcAddress = "bcrt1qujs29q4gkyn2uj6y570xl460p4y43ruayxu8ry";
+            const daveVaultId = api.createType("AccountId", dave.address);
+            await redeemAPI.request(redeemAmountAsSatoshi, btcAddress, daveVaultId);
```

Comment:
At the end of the test, should check if a suitable request redeem event has been fired.

Reply:
I'll add a check in the redeem API, same as for issue